consensus/state.go

```diff
@@ -1131,8 +1131,17 @@ func (cs *State) defaultDecideProposal(height int64, round int32) {
 		block, blockParts = cs.ValidBlock, cs.ValidBlockParts
 	} else {
 		// Create a new proposal block from state/txs from the mempool.
-		block, blockParts = cs.createProposalBlock()
-		if block == nil {
+		var err error
+		block, err = cs.createProposalBlock()
+		if err != nil {
+			cs.Logger.Error("unable to create proposal block", "error", err)
+			return
+		} else if block == nil {
+			return
+		}
```

Comment:
When does this happen?

Reply:
Good point. I checked the new logic of `createProposalBlock` and it seems it's not possible (either you get the block or an error).
I'll change the return by a panic.

Reply:
BTW, this is something I didn't catch as a reviewer last time (in #8094) 😉